system-addon/test/unit/lib/Store.test.js

```diff
@@ -38,55 +49,129 @@ describe("Store", () => {
+    });
+    it("should not override the pref if it was already set", () => {
+      Preferences.set(`${PREF_PREFIX}foo`, false);
+      store.maybeStartFeedAndListenForPrefChanges("foo");
+      assert.isFalse(Preferences.get(`${PREF_PREFIX}foo`));
+    });
+    it("should initialize the feed if the Pref is set to true", () => {
+      Preferences.set(`${PREF_PREFIX}foo`, true);
+      store.maybeStartFeedAndListenForPrefChanges("foo");
+      assert.calledWith(store.initFeed, "foo");
+    });
+    it("should not initialize the feed if the Pref is set to false", () => {
+      Preferences.set(`${PREF_PREFIX}foo`, false);
+      store.maybeStartFeedAndListenForPrefChanges("foo");
+      assert.notCalled(store.initFeed);
+    });
+    it("should observe the pref", () => {
+      store.maybeStartFeedAndListenForPrefChanges("foo");
+      assert.calledWith(Preferences.observe, `${PREF_PREFIX}foo`, store._prefHandlers.get(`${PREF_PREFIX}foo`));
+    });
+    describe("handler", () => {
+      let handler;
+      beforeEach(() => {
+        store.maybeStartFeedAndListenForPrefChanges("foo");
+        handler = store._prefHandlers.get(`${PREF_PREFIX}foo`);
+      });
+      it("should initialize the feed if called with true", () => {
+        handler(true);
+        assert.calledWith(store.initFeed, "foo");
+      });
+      it("should uninitialize the feed if called with false", () => {
+        handler(false);
+        assert.calledWith(store.uninitFeed, "foo");
+      });
+    });
+  });
+  describe("#init", () => {
+    it("should set ._feedConstructors to the object passed to init", () => {
+      class Foo {}
+      const feeds = {foo: () => new Foo(), bar: () => new Foo()};
+      store.init(feeds);
+      assert.equal(store._feedConstructors, feeds);
+    });
+    it("should call .maybeStartFeedAndListenForPrefChanges with each key", () => {
+      sinon.stub(store, "maybeStartFeedAndListenForPrefChanges");
+      store.init({foo: () => {}, bar: () => {}});
+      assert.calledWith(store.maybeStartFeedAndListenForPrefChanges, "foo");
+      assert.calledWith(store.maybeStartFeedAndListenForPrefChanges, "bar");
     });
     it("should initialize the MessageManager channel", () => {
       store.init();
       assert.calledOnce(store._mm.createChannel);
     });
   });
   describe("#uninit", () => {
-    it("should clear .feeds", () => {
-      store.init([{}, {}, {}]);
-      assert.equal(store.feeds.size, 3);
+    it("should clear .feeds, ._prefHandlers, and ._feedConstructors", () => {
+      store.init({
+        a: () => ({}),
+        b: () => ({}),
+        c: () => ({})
+      });
 
       store.uninit();
 
       assert.equal(store.feeds.size, 0);
-    });
-    it("should dispatch an uninit action", () => {
-      sandbox.spy(store, "dispatch");
-
-      store.uninit();
-
-      assert.calledWith(store.dispatch, {type: at.UNINIT});
+      assert.equal(store._prefHandlers.size, 0);
```

Comment:
In this case don't think it's really necessary, since `size` is a standard property of all `Set`s/`Map`s